### PR TITLE
Re-use httpclient for continuous queries

### LINKED ISSUE #2310: 0.9.0_rc25: all socket ports used up
When I install and start RC25 on my Ubuntu 14.04 box, I run out of local ports in a couple of minutes. As a result, other software isn't able to open TCP sockets.

I found one similar report [in the freenode IRC log](https://botbot.me/freenode/influxdb/2014-09-29/?msg=22535566&page=2).

    $ sudo apt-get purge influxdb
    Reading package lists... Done
    Building dependency tree       
    Reading state information... Done
    The following packages will be REMOVED:
      influxdb*
    0 upgraded, 0 newly installed, 1 to remove and 56 not upgraded.
    After this operation, 22.2 MB disk space will be freed.
    Do you want to continue? [Y/n] 
    (Reading database ... 385427 files and directories currently installed.)
    Removing influxdb (0.9.0-rc25) ...
    Purging configuration files for influxdb (0.9.0-rc25) ...
    dpkg: warning: while removing influxdb, directory '/opt/influxdb' not empty so not removed
    $ sudo rm -rf /opt/influxdb
    $ sudo dpkg -i influxdb_0.9.0-rc25_amd64.deb 
    Selecting previously unselected package influxdb.
    (Reading database ... 385416 files and directories currently installed.)
    Preparing to unpack .../influxdb_0.9.0-rc25_amd64.deb ...
    Unpacking influxdb (0.9.0-rc25) ...
    Setting up influxdb (0.9.0-rc25) ...
     Removing any system startup links for /etc/init.d/influxdb ...
       /etc/rc0.d/K20influxdb
       /etc/rc1.d/K20influxdb
       /etc/rc2.d/S20influxdb
       /etc/rc3.d/S20influxdb
       /etc/rc4.d/S20influxdb
       /etc/rc5.d/S20influxdb
       /etc/rc6.d/K20influxdb
     Adding system startup for /etc/init.d/influxdb ...
       /etc/rc0.d/K20influxdb -> ../init.d/influxdb
       /etc/rc1.d/K20influxdb -> ../init.d/influxdb
       /etc/rc6.d/K20influxdb -> ../init.d/influxdb
       /etc/rc2.d/S20influxdb -> ../init.d/influxdb
       /etc/rc3.d/S20influxdb -> ../init.d/influxdb
       /etc/rc4.d/S20influxdb -> ../init.d/influxdb
       /etc/rc5.d/S20influxdb -> ../init.d/influxdb
    $ while sleep 1; do netstat -tn|grep TIME_WAIT|wc --lines; done
    9
    9
    9
    9
    9
    9
    9
    9
    9^C
    $ sudo service influxdb start
    Starting the process influxd [ OK ]
    influxd process was started [ OK ]
    $ while sleep 1; do netstat -tn|grep TIME_WAIT|wc --lines; done
    10
    10
    10
    14
    18
    22
    31
    37
    67
    94
    128
    177
    236
    305
    353
    497
    632
    787
    1017
    1216
    1301
    1561
    1830
    2164
    2508
    2844
    3300
    3811
    4187
    4642
    4810
    5268
    5618
    5789
    6173
    6631
    6824
    7183
    7561
    7995
    8270
    8665
    9169
    9362
    9758
    10201
    10501
    10835
    11174
    11585
    11788
    12162
    12607
    12933
    13094
    13462
    13919
    14122
    14624
    14887
    15219
    15478
    15777
    16316
    16704
    16958
    17282
    17709
    18158
    18377
    18796
    19206
    18819
    19038
    19453
    19877
    20282
    20537
    20895
    21334
    21765
    22012
    22383
    22799
    21435
    21612
    22039
    22415
    22813
    22995
    23408
    23832
    24245
    24475
    25017
    25276
    25880
    23466
    23677
    24010
    24478
    24930
    25109
    25577
    25988
    25998
    25998
    25998
    23859
    24946
    24947
    24946
    24946
    24946
    24946
    24946
    24946
    22433
    22433
    22433
    22435
    22435
    22435
    22436
    23256
    20947
    21477
    21476
    21494
    21476
    21476
    21476
    19259
    22059
    23230
    23880
    24266
    24744
    24911
    24943
    24943
    24943
    24943
    21532
    24605
    24949
    25462
    25796
    25982
    25979
    25979
    25979
    25979
    25979
    23224
    23898
    25112
    25112
    25112
    25112
    25119
    25119
    23754
    25129
    25129
    25129
    25129
    25136
    25129
    25129
    23956
    23375
    23406
    23375
    23375
    23375
    23376
    23376
    23376
    23376
    23374
    23375
    23374
    23374
    22713
    22713
    22712
    22762
    22714
    23329
    23329
    17689
    20578
    20795
    21437
    21946
    22458
    22602^C
$

### PR DESCRIPTION
**Pull request by jwilder**
Fixes #2310

### CODE REVIEW

**Comment by otoolep** [comment]
Don't forget the CHANGELOG.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,7 +1,8 @@
 ## v0.9.0-rc26 [unreleased]
 
 ### Bugfixes
-- [#2297](https://github.com/influxdb/influxdb/pull/2297) -- create /var/run during startup. Thanks @neonstalwart.
+- [#2297](https://github.com/influxdb/influxdb/pull/2297): create /var/run during startup. Thanks @neonstalwart.
+- [#2310](https://github.com/influxdb/influxdb/pull/2310): all socket ports used up
```
Comment (otoolep) [inline CHANGELOG.md]: This is actually PR #2312. Can I suggest the message should be something like "Broker should re-use HTTP client when invoking CQs"?

**Comment by otoolep** [comment]
+1 on CHANGELOG fix-up.

**Comment by toddboom** [comment]
:+1:

**Pull request merged**